pkl-core/src/main/java/org/pkl/core/util/IoUtils.java

```diff
@@ -176,7 +176,7 @@ public static Path getDefaultModuleCacheDir() {
 
   // not stored to avoid build-time initialization by native-image
   public static String getLineSeparator() {
-    return System.getProperty("line.separator");
+    return System.lineSeparator();
```

Comment:
Per comment on line 177, this is likely problematic because `System` gets initialized at build time and added to the heap (we initialize everything at build time via the `--initialize-at-build-time=` flag).